policies/recipes/installrecipe.go

```diff
@@ -15,24 +15,35 @@
 package recipes
 
 import (
+	"bytes"
 	"context"
+	"crypto/sha256"
+	"encoding/hex"
 	"fmt"
 	"os"
 	"path/filepath"
 	"time"
 
 	osconfigpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha2"
+	"github.com/golang/protobuf/jsonpb"
 )
 
 const (
 	recipeBasePath = "/tmp/osconfig_software_recipes"
 	// TODO: move to constants, split linux and windows.
 )
 
+var (
+	marshaller = jsonpb.Marshaler{}
```

Comment:
why is it global?

Reply:
My impression is that it is just configuration for the marshaler, which shouldn't vary at all so the thought was just to initialize it once. While looking I can't find any safety guarantees for it so I'll change it to creating a new one each time as the empty struct allocation should be negligible cost.